src/resources/rmt_esp32.cc

```diff
@@ -256,12 +264,42 @@ PRIMITIVE(transmit) {
   ARGS(RMTResource, resource, Blob, items_bytes)
   if (items_bytes.length() % 4 != 0) INVALID_ARGUMENT;
 
+  // We are going to pass a pointer to a C function which will consume it, while we
+  // yield back to the Toit VM. There could be GCs while the C function uses the memory.
+  // As such, we need an external address.
+  const uint8* address = items_bytes.address();
+  Object* keep_alive = _raw_items_bytes;
+  if (_raw_items_bytes->is_byte_array() && ByteArray::cast(_raw_items_bytes)->has_external_address()) {
+    // Nothing to do. We already have an external address.
+  } else {
+    // Create an external byte array with the same size.
+    // We will return it to the caller, so they can keep it alive.
+    Error* error = null;
+    // Force external.
+    ByteArray* external_copy = process->allocate_byte_array(items_bytes.length(), &error, true);
+    if (external_copy == null) return error;
+    ByteArray::Bytes bytes(external_copy);
+    memcpy(bytes.address(), address, items_bytes.length());
+    address = bytes.address();
+    keep_alive = external_copy;
+  }
+
   rmt_channel_t channel = resource->channel();
-  const rmt_item32_t* items = reinterpret_cast<const rmt_item32_t*>(items_bytes.address());
-  esp_err_t err = rmt_write_items(channel, items, items_bytes.length() / 4, true);
-  if ( err != ESP_OK) return Primitive::os_error(err, process);
+  const rmt_item32_t* items = reinterpret_cast<const rmt_item32_t*>(address);
+  bool wait_until_done;
```

Comment:
bool wait_until_done;  // Local for naming argument to rmt_write_items.

Reply:
done.